classes/DirtEvent.sc

```diff
@@ -100,6 +100,7 @@ DirtEvent {
 				delta * ~legato.value
 			} {
 				unitDuration = unitDuration ? delta;
+				if (~timescale.notNil) {unitDuration = unitDuration * ~timescale };
```

Comment:
I think this is not the right place to add the stretch. Better do it when the sustain has been fully calculated. We'll need to carefully test this.

```supercollider
		if (~timescale.notNil) { sustain = sustain * ~timescale };
		~sustain = sustain - (~fadeTime + ~fadeInTime);

```